### ISSUE TITLE
matnwb does not support spaces in names

### ISSUE THREAD

**Issue by bendichter** [NeurodataWithoutBorders/matnwb#35]
Since names are used as field names of objects, matnwb does not allow them to have spaces. pynwb does allow spaces. Maybe a compromise would be to automatically replace all spaces with underscores in matnwb.

**Comment by ln-vidrio** [comment]
I believe spaces work with open Sets (basically, any Group structure that can have 0 or more items in it) as they are Map-backed.  The only other place that could include spaces would be at the YAML level (core schema and extensions) which will definitely require some work.

**Comment by bendichter** [comment]
Yes, the problem I was having with specifically with the names of extensions, but I think the best solution would be to enforce no spaces on the `pynwb` side

**Comment by ln-vidrio** [comment]
So I see two thing needing work to get extensions with spaces working:
1. Add a check when generating files to replace spaces in property values.  This should be fairly simple.
2. Do a check before reading a compound data type, recreate the compound type using new member names, then reading it as normal.

This will "allow" spaces for class properties and makes compound data types readable if they have spaces.

I'll look into this.

**Comment by ehennestad** [comment]
@bendichter Is this still relevant? 

If yes, can you clarify what is meant here:
> Since names are used as field names of objects
Does the schema allow for spaces in object property names? 

Could you also point me to an example scenario which illustrates this problem?

**Comment by bendichter** [comment]
@ehennestad
Yes, this issue is still relevant. Let me explain with a concrete example:

When creating a trials table in MatNWB, you use dot notation like this:
```matlab
nwbfile.trials = DynamicTable(...)
```

This syntax pattern appears whenever you're working with Datasets or Containers in the schema that have set names. While this works fine for the core schema (which doesn't have this issue), we run into a potential problem when set names contain spaces.

The issue doesn't occur in the core schema, but it's not currently enforced as a rule, meaning it could appear in extensions. If an extension defines a Container or Dataset that has a sub-Container or Dataset with a space in its set name, we hit a compatibility problem.

There are two potential solutions we could consider:
1. Enforce a rule that prohibits spaces in set names
2. Implement a workaround in MatNWB (for example, automatically replacing spaces with underscores)

While I can't recall the specific instance where we encountered this, it remains a potential issue that we should address.
